pango/src/attr_list.rs

```diff
@@ -27,18 +27,18 @@ impl AttrList {
     }
 
     #[doc(alias = "pango_attr_list_insert")]
-    pub fn insert(&self, attr: impl IsAttribute) {
+    pub fn insert(&self, attr: impl AsRef<Attribute>) {
         unsafe {
-            let attr = attr.upcast();
+            let attr = attr.as_ref();
             ffi::pango_attr_list_insert(self.to_glib_none().0, attr.to_glib_none().0 as *mut _);
             mem::forget(attr); //As attr transferred fully
```

Comment:
is the mem::forget still needed here?

Reply:
or reworded: how should i handle the clippy warning because now i have a reference

Reply:
Well you would have to copy the attribute now :) It's better how the function was before because then it took actual ownership of the value and passed on ownership to the C function.

Reply:
indeed, but then we have to manually cast when adding an attribute to the AttrList

Reply:
Why? `attr: impl IsAttribute` gives you ownership

Reply:
librsvg has a use case where it creates multiples attributes; stores them on a `Vec` then iterate the attributes to set a start_index/end_index; only after that it adds them to the AttrList. Which gives you `Attribute` in the Vec & you can't make Attribute implement IsAttribute

Reply:
I don't know, you either need to copy or pass by value one way or another. There's no third way :)

Reply:
Copying it is then :)